aten/src/ATen/native/transformers/cuda/flash_attn/fmha_api.cpp

```diff
@@ -64,7 +64,7 @@ void set_params_fprop(FMHA_fprop_params &params,
                       bool is_causal) {
 
     // Reset the parameters
-    memset(&params, 0, sizeof(params));
+    params = {};
```

Comment:
Hmm, technically those are not equivalent, are they?  I.e. default constructor might not set the value for all parameters or might set it to values other than 0?

Reply:
Correct, these are not directly equivalent. memset is not allowed here though.

The rules for this are somewhat complicated, but most fields should be zeroed out with this. Nevertheless, most fields are written directly in this function anyways, with the exception of those that are called out in the summary above.